### DISCUSSION TITLE
r.runCommandWithSelectionOrWord availability

### DISCUSSION THREAD

**Discussion by MikeLydeamore** [posit-dev/positron#5747] [Q&A] [answered]
I have used the `r.runCommandWithSelectionOrWord` pretty extensively in VSCode, mainly for working with `targets`. Is there an equivalent in positron? 

I know I can use `workbench.action.executeCode.console` but I don't seem to be able to pass in a highlighted word into the command.

**Comment by nstrayer** [comment]
We dont currently have this functionality, but you inspired me and I made a (draft) PR for this. 
#5784 
It's pretty similar to the functionality that we have with running the selection with the keyboard shortcut. You want it as a command so you can script it?

**Comment by nstrayer** [comment, marked as answer]
Actually I was incorrect. We _do_ have this already, no need for my PR! The command is
`workbench.action.positronConsole.executeCode`. This is the command that gets run with the command + enter keyboard shortcut!

**Reply by MikeLydeamore** [reply]
Is there a way for me to include the highlighted text in the command? For example, I'd like to do

```targets::tar_load(selected_symbol)```

My understanding of `workbench.action.positronConsole.executeCode` is that I can't use it as an argument to a function.

With the vscode-R extension I can use

```
"command": "r.runCommandWithSelectionOrWord",
"when": "editorTextFocus",
"args": "targets::tar_load($$)"
```

**Reply by jennybc** [reply]
We don't currently expose `workbench.action.executeCode.console` in way that "just works" for interpolating text, like the current selection or whatnot.

As an experiment, I put this in my `keybindings.json` file:

```
[ ...
{
    "key": "Cmd+Shift+S",
    "command": "workbench.action.executeCode.console",
    "when": "editorTextFocus",
    "args": {
        "langId": "r",
        "code": "str(${selectedText})",
        "focus": true
    }
},
...]
```

But `$selectedText` does not get substituted into a custom command like this. If I selected the name of an object and execute the above, I see:

```
> str(${selectedText})
Error:
! unexpected '$' in "str($"
```

But I think this is along the lines of what you want, yeah? You want to be able to use something like this variable substitution mechanism with `executeCode.console`:

https://code.visualstudio.com/docs/editor/variables-reference

**Reply by MikeLydeamore** [reply]
Yes, that's right. 

My understanding of the variable substitution page is that it only works in tasks, not keybindings though, which is why the substitution isn't working.

**Reply by jennybc** [reply]
I know. I just wanted to get very concrete about the ask.

**Comment by juliasilge** [comment]
I opened https://github.com/posit-dev/positron/issues/5797 to track and prioritize this feature request. Thanks @MikeLydeamore! 🙌

**Reply by mskyttner** [reply]
Just wanted to say that it would be great to be able to select text in a .sql file and use a keyboard shortcut (like Shift+Enter or something that doesn't clash with Ctrl+Enter) to send it to the Terminal (not Console) where for example the duckdb prompt would execute whatever is sent there.

I tried to set it up hoping to get some hints for the sql dialect but the vscode instructions (files.associations) I was looking at didn't seem to work in Positron

```json
"files.associations": {
    "*.sql": "sql"
}
```
I think I succeeded with making a setting like this but it appears the "workbench.action.terminal" isn't really the Terminal but the Console:

```json
{
  "key": "shift+enter",
  "command": "workbench.action.terminal.sendSequence",
  "args": { "text": "${selectedText}\\n" },
  "when": "editorTextFocus && terminalFocus"
}
```

**Reply by juliasilge** [reply]
@mskyttner can you add some of these details of what you'd like to do to #5251? It's a bit different than the original request here in this discussion (terminal vs. console).
